circuitbreaker/circuit_breaker.go

```diff
@@ -54,82 +56,81 @@ type CircuitBreaker struct {
 }
 
 func NewCircuitBreaker(config Config) *CircuitBreaker {
+	if config.CircuitName == "" {
+		return nil
```

Comment:
Could we return `error` in this case (cause there is no place where we check if the created instance is nil or not)?